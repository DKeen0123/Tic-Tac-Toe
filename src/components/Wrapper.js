import React, { Component } from 'react';
import Header from './Header';
import PlayerTurn from './PlayerTurn';
import Arena from './Arena';
import Box from './Box';

class Wrapper extends Component {
  constructor() {
    super();

    this.state = {
      playerOneTurn: true,
      boxes: []
    };
  }
  componentWillMount() {
    this.populateBoxes();
  }

  populateBoxes() {
    let { boxes } = this.state;
    let numbers = Array.apply(null, { length: 9 }).map(Number.call, Number);

    numbers.map(number =>
      boxes.push(
        <Box
          key={number}
          id={number}
          handlePlayerMove={this.handlePlayerMove}
        />
      )
    );

    this.setState({ boxes });
  }

  handlePlayerMove = event => {
    let { playerOneTurn } = this.state;
    this.setState({ playerOneTurn: !playerOneTurn });
  };

  render() {
    return (
      <div>
        <Header />
        <PlayerTurn />
        <Arena boxes={this.state.boxes} />
      </div>
    );
  }
}

export default Wrapper;
