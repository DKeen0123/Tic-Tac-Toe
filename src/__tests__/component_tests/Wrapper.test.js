import React from 'react';
import { shallow } from 'enzyme';
import Wrapper from '../../components/Wrapper';

describe('Wrapper', () => {
  const wrapper = shallow(<Wrapper />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a header component', () => {
    expect(wrapper.find('Header').exists()).toBe(true);
  });
});
