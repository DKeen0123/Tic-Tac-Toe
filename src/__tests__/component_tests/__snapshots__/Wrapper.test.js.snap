// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wrapper />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <PlayerTurn />,
        <Arena
          boxes={
                    Array [
                              <Box
                                handlePlayerMove={[Function]}
                                id={0}
                    />,
                              <Box
                                handlePlayerMove={[Function]}
                                id={1}
                    />,
                              <Box
                                handlePlayerMove={[Function]}
                                id={2}
                    />,
                              <Box
                                handlePlayerMove={[Function]}
                                id={3}
                    />,
                              <Box
                                handlePlayerMove={[Function]}
                                id={4}
                    />,
                              <Box
                                handlePlayerMove={[Function]}
                                id={5}
                    />,
                              <Box
                                handlePlayerMove={[Function]}
                                id={6}
                    />,
                              <Box
                                handlePlayerMove={[Function]}
                                id={7}
                    />,
                              <Box
                                handlePlayerMove={[Function]}
                                id={8}
                    />,
                            ]
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "boxes": Array [
            <Box
              handlePlayerMove={[Function]}
              id={0}
/>,
            <Box
              handlePlayerMove={[Function]}
              id={1}
/>,
            <Box
              handlePlayerMove={[Function]}
              id={2}
/>,
            <Box
              handlePlayerMove={[Function]}
              id={3}
/>,
            <Box
              handlePlayerMove={[Function]}
              id={4}
/>,
            <Box
              handlePlayerMove={[Function]}
              id={5}
/>,
            <Box
              handlePlayerMove={[Function]}
              id={6}
/>,
            <Box
              handlePlayerMove={[Function]}
              id={7}
/>,
            <Box
              handlePlayerMove={[Function]}
              id={8}
/>,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <PlayerTurn />,
          <Arena
            boxes={
                        Array [
                                    <Box
                                      handlePlayerMove={[Function]}
                                      id={0}
                        />,
                                    <Box
                                      handlePlayerMove={[Function]}
                                      id={1}
                        />,
                                    <Box
                                      handlePlayerMove={[Function]}
                                      id={2}
                        />,
                                    <Box
                                      handlePlayerMove={[Function]}
                                      id={3}
                        />,
                                    <Box
                                      handlePlayerMove={[Function]}
                                      id={4}
                        />,
                                    <Box
                                      handlePlayerMove={[Function]}
                                      id={5}
                        />,
                                    <Box
                                      handlePlayerMove={[Function]}
                                      id={6}
                        />,
                                    <Box
                                      handlePlayerMove={[Function]}
                                      id={7}
                        />,
                                    <Box
                                      handlePlayerMove={[Function]}
                                      id={8}
                        />,
                                  ]
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "boxes": Array [
              <Box
                handlePlayerMove={[Function]}
                id={0}
/>,
              <Box
                handlePlayerMove={[Function]}
                id={1}
/>,
              <Box
                handlePlayerMove={[Function]}
                id={2}
/>,
              <Box
                handlePlayerMove={[Function]}
                id={3}
/>,
              <Box
                handlePlayerMove={[Function]}
                id={4}
/>,
              <Box
                handlePlayerMove={[Function]}
                id={5}
/>,
              <Box
                handlePlayerMove={[Function]}
                id={6}
/>,
              <Box
                handlePlayerMove={[Function]}
                id={7}
/>,
              <Box
                handlePlayerMove={[Function]}
                id={8}
/>,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
